### PR TITLE
refactor(common): #345 - Extract RelativeCommonMap to separate api package

### PR DESCRIPTION
**Pull request by grzesuav**
Second part of extracting, this time common map object used to storing relative members.

### CODE REVIEW

File: pkg/controller/common/api/v1/common.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// RelativeObjectMap holds object related to given parent object.
+// The structure is [GroupVersionKind] -> [common.relativeName] -> *unstructured.Unstructured
+// where:
+//   GroupVersionKind - identifies type stored in entry
+//   relativeName() - return path to object in relation to parent
```
Comment (AmitKumarDas) [inline pkg/controller/common/api/v1/common.go:16]: nit: can we provide a sample/example here

Comment (AmitKumarDas) [reply pkg/controller/common/api/v1/common.go:16]: query: Has namespace anything to do w.r.t this map?

File: pkg/controller/common/api/v1/common.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// RelativeObjectMap holds object related to given parent object.
```
Comment (AmitKumarDas) [inline pkg/controller/common/api/v1/common.go:12]: disclaimer: I understand this is refactoring of existing packages. However provided some nits & suggestions at some places.

File: pkg/controller/common/api/v1/common.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	if parent.GetNamespace() == "" && obj.GetNamespace() != "" {
+		return fmt.Sprintf("%s/%s", obj.GetNamespace(), obj.GetName())
+	}
+	return obj.GetName()
+}
+
+// ReplaceObject replaces the object with the same name & namespace as
+// the given object with the contents of the given object. If no object exists
+// in the existing map then no action is taken.
+func (m RelativeObjectMap) ReplaceObject(parent v1.Object, obj *unstructured.Unstructured) {
```
Comment (AmitKumarDas) [inline pkg/controller/common/api/v1/common.go:76]: nit: Is it possible to rename this to MaybeReplaceObject or ReplaceObjectIfExists

File: pkg/controller/composite/hooks_test.go [hunk 4]
```diff
@@ -2,7 +2,7 @@ package composite
 
 import (
 	"metacontroller/pkg/apis/metacontroller/v1alpha1"
-	"metacontroller/pkg/controller/common"
+	v12 "metacontroller/pkg/controller/common/api/v1"
```
Comment (AmitKumarDas) [inline pkg/controller/composite/hooks_test.go:5]: IDE effect?

Comment (grzesuav) [reply pkg/controller/composite/hooks_test.go]: yes ;)

File: pkg/controller/common/api/v1/common.go (hunk 3 above)
Comment (grzesuav) [reply pkg/controller/common/api/v1/common.go]: sure

File: pkg/controller/common/api/v1/common.go (hunk 2 above)
Comment (grzesuav) [reply pkg/controller/common/api/v1/common.go:12]: yes, thank you

File: pkg/controller/common/api/v1/common.go (hunk 1 above)
Comment (grzesuav) [reply pkg/controller/common/api/v1/common.go:16]: yes, it depends also on parent type

**Review by AmitKumarDas — APPROVED** [review]

**Pull request merged**